sonar-text-plugin/src/main/java/org/sonar/plugins/secrets/api/SpecificationBasedCheck.java

```diff
@@ -59,6 +59,10 @@ public void analyze(InputFileContext ctx) {
     List<TextRange> foundSecrets = new ArrayList<>();
 
     for (SecretMatcher secretMatcher : matcher) {
+      if (!secretMatcher.getPreFilter().test(ctx)) {
+        continue;
+      }
```

Comment:
IMO this logic should reside in the secretMatcher. Maybe we should change `secretMatcher.findIn(...)` to accept InputFileContext instead of the content